nixos/modules/services/web-servers/stargazer.nix

```diff
@@ -204,11 +204,9 @@ in
     };
 
     # Create default cert store
-    system.activationScripts.makeStargazerCertDir =
-      lib.optionalAttrs (cfg.store == /var/lib/gemini/certs) ''
-        mkdir -p /var/lib/gemini/certs
-        chown -R ${cfg.user}:${cfg.group} /var/lib/gemini/certs
-      '';
+    systemd.tmpfiles.rules = [
+      ''d /var/lib/gemini/certs - "${cfg.user}" "${cfg.group}" -''
```

Comment:
This has the slight behaviour difference of not changing the ownership of the _contents_. It's probably more correct, and hopefully won't break anything in practice, but maybe @gaykitty could double-check this?

Reply:
Could add a `Z` line to recursively change the mode and ownership.

Reply:
The recursive chown should not be necessary as any files created in that directory should already have the correct permissions. This change appears to remove the conditionality of that directory being created, I'd prefer to maintain that if possible. 

I'm not familiar with `systemd.tmpfiles.rules` but it's name and description would suggest it's for *temporary* files, which these cert are not. They should **never** be automatically deleted. Can someone clarify the behavior of this directive for me?

Reply:
Oh right I missed the condition. Readded it. tmpfiles is a poorly chosen name because it can create any file and that's what we are doing here. There is no automatic cleanup.

Reply:
Sounds good! Then this all seems a okay to me.

Reply:
@nikstur I'd still recommend a `Z` line unless @gaykitty advises against it. It just makes sure the whole directory is owned by the right  user.